### PR TITLE
D8CORE-1401: wysiwyg tweaks

### PR DESCRIPTION
**Pull request by cjwest**
# READY FOR REVIEW

# Summary
- fixes issues with WYSIWYG styles

# Needed By (Date)
- Soon

# Urgency
- Fixes broken stuff in the WYSIWYG

# Steps to Test

1. Pull this branch
2. clear caches
3. verify styles in the WYSIWYG:
```
p {
  padding-bottom: 3.4rem; 
  margin-bottom: 0px; 
} 
And for the H2 + H3:

h2, h3 {
  margin-bottom: 0; /*as in the modular spacer*/
}
Callout:

.su-callout-text {
  margin-top: 0; /*as in the modular spacer*/
}
```
Additionally, 
Verify that the spacing is correct for the "related" or "Card" WYSIWYG style.
It fixes the padding when this style is first:

Screen Shot 2020-03-06 at 4 33 11 PM
 Versus:

Screen Shot 2020-03-06 at 4 34 48 PM

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket: https://stanfordits.atlassian.net/browse/D8CORE-1401
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)
@kerri-augenstein 

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: modules/stanford_profile_styles/lib/scss/paragraph-types/wysiwyg/_paragraph.scss [hunk 1]
```diff
@@ -4,8 +4,8 @@
 
   p {
     @include big-paragraph;
-    @include modular-spacing('margin-bottom', 1);
-    @include margin(0 null null null);
+    @include margin(0 null 0 null);
+    @include modular-spacing('padding-bottom', 2);
```
Comment (sherakama) [inline modules/stanford_profile_styles/lib/scss/paragraph-types/wysiwyg/_paragraph.scss]: This is mixing padding and margin again. Shouldn't we stick to one to be more predictable?

Comment (cjwest) [reply modules/stanford_profile_styles/lib/scss/paragraph-types/wysiwyg/_paragraph.scss]: That's a great question. I was following what @kerri-augenstein  had specified in this ticket. 
https://stanfordits.atlassian.net/browse/D8CORE-1401

Comment (sherakama) [reply modules/stanford_profile_styles/lib/scss/paragraph-types/wysiwyg/_paragraph.scss]: We are using margin everywhere else. Stick to using margin on this one, please.

Comment (cjwest) [reply modules/stanford_profile_styles/lib/scss/paragraph-types/wysiwyg/_paragraph.scss]: @sherakama in the ticket https://stanfordits.atlassian.net/browse/D8CORE-1401, Kerri has specifically asked to use margin.

> I think we want to kill the margin applied to our <p>, and instead opt for padding only, as the margin on the <p> is overlapping with the margins applied to headings, and we want them to stack.

Comment (sherakama) [reply modules/stanford_profile_styles/lib/scss/paragraph-types/wysiwyg/_paragraph.scss]: Please stick to using margin only. Both padding and margin are accomplishing the same thing in this case and I'd rather stay consistent with our other elements.

**Comment by cjwest** [comment]
@sherakama,
I updated this based on our conversation yesterday where we decided to replace margin-top with padding-top, and remove any padding-bottom (except where required such as the related-text).

New but related topic:
In reviewing how the margins are applied, I've noticed that on the buttons we're applying the margin to the `<a>` tag that is within the `<p>` tag. Whereas on the other styles, we're applying the margin to the margin to the `<p>` tag. 

That's a lot of words. ⏫ 

Here's the margin on the `<a>`:
Screen Shot 2020-03-11 at 12 08 11 PM

Here's the margin on the `<p>`
Screen Shot 2020-03-11 at 12 08 19 PM

Here's the code:

```
<p>
  <a class="su-button--secondary" href="http://www.stanford.edu">Secondary Button Style</a>
</p>
```

I wonder if it would make it more consistent to apply the margin to the <p> tag for buttons as well?

**Comment by sherakama** [comment]
@cjwest In regards to the issue you brought up with the button margins please open up a new ticket and PR to continue this conversation. I think it will need to be put through its paces with other content around it.

**Comment by sherakama** [comment]
Also, thank you for your hard work on this issue.

**Pull request merged**